src/librustc/mir/tcx.rs

```diff
@@ -16,8 +16,7 @@ pub enum PlaceTy<'tcx> {
     Ty { ty: Ty<'tcx> },
 
     /// Downcast to a particular variant of an enum.
-    Downcast { adt_def: &'tcx AdtDef,
-               substs: SubstsRef<'tcx>,
+    Downcast { ty: Ty<'tcx>,
                variant_index: VariantIdx },
```

Comment:
Since we are going ahead with this, might as well switch to:
```rust
pub struct PlaceTy<'tcx> {
    pub ty: Ty<'tcx>,
    pub variant_index: Option<VariantIdx>,
}
```

Reply:
done